src/aasthaJJun20/ArraysUnionIntersection.java

```diff
@@ -0,0 +1,80 @@
+package aasthaJJun20;
+
+public class ArraysUnionIntersection {
+	// This method is written just to display the array elements on console
+	private void displayArray(double[] input) {
+		System.out.print("{ ");
+		for (int index = 0; index < input.length; index++) {
+			System.out.print(input[index] + "   ");
+		}
+		System.out.println("}");
+	}
+
+	// Below method is written to search a particular element in the provided(input)
+	// array
+	private boolean searchInArray(double[] input, double element) {
+		for (int index = 0; index < input.length; index++) {
+			if (input[index] == element)
+				return true;
+		}
+		return false;
+	}
+
+	// Program 1: Find union of two arrays.
+	public double[] getUnionOfArray(double[] arr1, double[] arr2) {
+		double[] unionArray = new double[arr1.length + arr2.length];
+		int size = 0;
+		for (int index = 0; index < arr1.length; index++) {
+			if (searchInArray(unionArray, arr1[index]) == false) {
+				unionArray[size] = arr1[index];
+				size++;
+			}
+		}
+		for (int index = 0; index < arr2.length; index++) {
+			if (searchInArray(unionArray, arr2[index]) == false) {
+				unionArray[size] = arr2[index];
+				size++;
+			}
+		}
+		double[] unionArrayResult = new double[size];
+		for (int index = 0; index < size; index++) {
+			unionArrayResult[index] = unionArray[index];
+		}
+		return unionArrayResult;
+	}
+
+	// Program 2: Find intersection(common elements) of two arrays.
+	public double[] getIntersectionOfArray(double[] arr1, double[] arr2) {
+		int size = 0;
+		if (arr1.length > arr2.length)
+			size = arr1.length;
+		else
+			size = arr2.length;
+		double[] intersectionArray = new double[size];
+		size = 0;
+		for (int index = 0; index < arr2.length; index++) {
+			if (searchInArray(arr1, arr2[index])) {
```

Comment:
if (searchInArray(unionArray, arr1[index],arr1.length-index) == false) {

think about it :)

Reply:
Here, When I'm passing **arr1.length-index** parameter, it is leaving the last element of arr2(which is common in both arrays).
when tried with just **arr1.length**, the output looks fine.

Reply:
Thats correct, you got it.
We can think of minimizing iteration in case of intersection. Here you used same method in both cases (thats actually nice) so have to think generalized solution.